### PR TITLE
[TNH_006]listing_image_upload

### PR DESCRIPTION
**Pull request by JakeGiangan**
[Commands]
`rails active_storage:install`
`rails db:migrate:reset db:seed`
`bundle install`

`rails s`

email: jake@email.com
password: password

[Pre-conditions]
1.)  rails active_storage:install
2.) rails db:migrate:reset db:seed
3.) create a new listing

[Expected Output]
1.) User can now add,  delete Listing Images  under a new listing

[Notes]
1.) If there's no image uploaded the carousel will not appear and only display the text "Not images found"

### CODE REVIEW

File: app/helpers/images_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module ImagesHelper
```
Comment (charlotteds) [inline app/helpers/images_helper.rb]: Delete unused file.

File: app/controllers/listings_controller.rb [hunk 2]
```diff
@@ -57,6 +59,10 @@ def destroy
     redirect_to properties_url
   end
 
+  def delete_image
+    
+  end
+
```
Comment (charlotteds) [inline app/controllers/listings_controller.rb]: Please complete the delete method on this task.

Comment (JakeGiangan) [reply app/controllers/listings_controller.rb]: Delete is already working and its in the images_controller I forgot to clear this one out during testing

File: app/controllers/listings_controller.rb [hunk 3]
```diff
@@ -75,6 +77,7 @@ def listing_params
       :has_heating,
       :has_air_conditioning,
       :address,
+      images: []
```
Comment (charlotteds) [inline app/controllers/listings_controller.rb]: This might not be a necessary parameter here.

**Review by charlotteds — COMMENTED** [review]
- I got the following error when trying to access Listing Details page. Please add a condition for rooms with no images.
Screen Shot 2021-03-24 at 6 24 31 PM
 
- I should be able to update listings from this page.
Screen Shot 2021-03-24 at 6 25 09 PM

File: app/views/listings/index.html.erb [hunk 4]
```diff
@@ -33,8 +37,11 @@
                                 </div>
                             </div>
                             <div class="row">
-                                <div class="col">
-                                    <%= link_to raw("View Details"), item, class: "btn btn-outline-success d-xl-flex justify-content-xl-center" %>
+                                <div class="col  d-xl-flex justify-content-xl-center">
+                                    <div class="btn-group" role="group">
+                                    <%= link_to raw("View Details"), item, class: "btn btn-outline-primary d-xl-flex justify-content-xl-center" %>
+                                    <%= link_to raw("Edit Details"), room_listing_path(item), class: "btn btn-outline-info d-xl-flex justify-content-xl-center" %>
+                                    </div>
```
Comment (charlotteds) [inline app/views/listings/index.html.erb]: Indent div contents

File: app/views/listings/show.html.erb [hunk 5]
```diff
@@ -1,6 +1,18 @@
...
+    background-attachment: fixed;
+    background-position: center;
+    background-repeat: no-repeat;
+    background-size: cover;'> 
+    </div>
+<% else %>
+    <div> 
+    <h1 class="text-danger d-xl-flex justify-content-xl-center" style="margin-top: 250px;margin-bottom: 150px;"> No Image uploaded! </h1>
+    </div>
+<% end %>
```
Comment (charlotteds) [inline app/views/listings/show.html.erb]: Indent div contents

**Review by charlotteds — COMMENTED** [review]
Listing should not be "publishable" when photos have not yet been added.
image

**Review by charlotteds — APPROVED** [review]

**Pull request merged**